crepecake/src/main/java/com/google/cloud/tools/crepecake/image/Image.java

```diff
@@ -16,29 +16,33 @@
 
 package com.google.cloud.tools.crepecake.image;
 
+import java.util.ArrayList;
 import java.util.Collections;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
 /** Represents an image. */
 public class Image {
 
   /** The layers of the image, in the order in which they are applied. */
   private ImageLayers<Layer> layers = new ImageLayers<>();
 
-  /** Environment variables for running the image. Maps from variable name to its value. */
-  private final Map<String, String> environmentMap = new HashMap<>();
+  /** Environment variable definitions for running the image, in the format {@code NAME=VALUE}. */
+  private final List<String> environment = new ArrayList<>();
```

Comment:
why was this flipped? Are we defining our json different now?

Reply:
This was changed because it is easier to just use the `NAME=VALUE` string rather than convert to-and-from a Map (i.e. existing `NAME=VALUE` strings do not need to be modified, and new `NAME=VALUE` strings that may be added can just be added as their string form).

Reply:
This is the sort of thing I've found Guava is really good at if you know where to look. In particular see Splitter and Joiner. 

Also it suggests maybe we're missing an abstraction. I.e. it probably shouldn't be a List or a Map, but rather some sort of `Environment` class that hides these details from the client.

Reply:
Filed https://github.com/GoogleCloudPlatform/minikube-build-tools-for-java/issues/91 to do this abstraction.